sdk/digitaltwins/mgmt-v2020_12_01/src/test/java/com/microsoft/azure/management/digitaltwins/v2020_12_01/DigitalTwinsLifecycleTests.java

```diff
@@ -0,0 +1,132 @@
+package com.microsoft.azure.management.digitaltwins.v2020_12_01;
+
+import com.microsoft.azure.PagedList;
+import com.microsoft.azure.arm.core.TestBase;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.CheckNameResultInner;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.DigitalTwinsDescriptionInner;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.DigitalTwinsManager;
+import com.microsoft.azure.management.digitaltwins.v2020_12_01.implementation.OperationInner;
+import com.microsoft.azure.management.resources.ResourceGroup;
+import com.microsoft.azure.management.resources.fluentcore.arm.Region;
+import com.microsoft.rest.RestClient;
+import com.microsoft.azure.management.resources.implementation.ResourceManager;
+import org.junit.Assert;
+import org.junit.Test;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+
+public class DigitalTwinsLifecycleTests extends TestBase {
+
+    protected ResourceManager resourceManager;
+    protected DigitalTwinsManager digitalTwinsManager;
+    private static String rgName;
+    protected String domain;
+    private static String defaultInstanceName = "DigitalTwinsSdk";
+    private static String defaultRegion = Region.US_WEST_CENTRAL.toString();
+    private static String defaultResourceGroupName = "rg2b9842374ecf6";
```

Comment:
I wanted to use the automatic RG name generator, but the test playback fails to replace it with every run, so I hardcoded a value. I will sync up with the SDK team on why the random rg name doesn't work with the test playback.

Reply:
If you use random name, you probably need to do it in the test method, not in class variable. The reason is that the name need to be recorded in the json, which happens during the test method.